### ISSUE TITLE
[request] working CH Layout

### ISSUE THREAD

**Issue by C0N5T-RUCT0R** [hak5/bashbunny-payloads#470]
I don't know what layout the ch layout is, but it doesn't work for me. I'am working with an QWERTZ CH-DE Layout. CAn you fix this or ad a new Layout please?

**Comment by githubkuyaya** [comment]
I can reproduce this. The characters `1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, `0`, `.` are a numpad issue. It should work fine if you activate the numlock key. Obviously, this is not intended behaviour and it should prefer the normal keys over the numpad ones. Maybe it's better to remove them from the language file, I don't know. I'll research a bit about this.
The other two characters, which did not work as expected, were `$` and `{`. These just had wrong keycodes, I don't know why. Maybe the layout changed or I just typed it in wrong. I'll provide a fix in a PR.
These were all the wrong characters, right?

**Comment by YvanLeRouge** [comment]
Hi C0N5T-RUCT0R,
I encounter the same problems as you, except that you managed to make the symbol \ (AltGr key + ><\)
In the result.txt we see that the \ is present while in the script there is only one space... I couldn't reproduce that!
Can you tell me how you manage to do that?
Thanks !!!
Sorry for my bad english.

**Comment by shr3d-d3r** [comment]
Hi,
Well, I didn't use the numpad. I went through every line of my Keyboard. First without shift or alt-ground, then with shift and than with alt-grnd.

You are right, I have forgotten to add \ at the last line.

**Comment by C0N5T-RUCT0R** [comment]
Hi,
Well, I didn't use the numpad. I went through every line of my Keyboard. First without shift or alt-ground, then with shift and than with alt-grnd.

You are right, I have forgotten to add \ at the last line.

**Comment by YvanLeRouge** [comment]
Thank you,
I didn't use the numpad either, but it's still crazy that I can't write the \ with my bash bunny and its "ch" language...
impossible to write a simple command like C:\ etc... I'm desperate..
Are you getting there?

**Comment by githubkuyaya** [comment]
No, the BashBunny uses the Numpad. It preferres the keycodes of the numbers on the numpad over the keycodes of the numbers located on the second top row of the keyboard. If you instruct the BashBunny to print `1`, it will simulate a `1` press on the numpad. Thus, for it to work, the Numlock key needs to be activated.
As I already stated above, I think the best way to work around that is to remove the Numpad keycodes from the language file.

**Comment by YvanLeRouge** [comment]
at githubkuyaya:
For me the numbers work well, only these two symbols \ { does not print...
An idea?

**Comment by bad-antics** [comment]
This is a well-documented issue with the Swiss-German (`ch.json`) HID language file. The root causes are:

### 1. Numpad vs. Regular Key Priority
The language file maps digits to **numpad keycodes** instead of the number row. This means NumLock must be ON for numbers to work, which is not a safe assumption on many systems.

**Fix**: Replace numpad keycodes with number row keycodes:
```json
"1": "00,00,1e",
"2": "00,00,1f",
"3": "00,00,20",
...
"0": "00,00,27"
```
Instead of the current numpad codes (`00,00,59` through `00,00,62`).

### 2. Backslash `\\` via AltGr
On Swiss-German layout, `\\` is produced with `AltGr + <` (the key left of Y). The HID mapping should be:
```json
"\\\\": "40,00,64"
```
Where `0x40` = Right Alt modifier, `0x64` = the `<>` key.

### 3. Curly brace `{`
`{` on CH layout is `AltGr + ü` (keycode `0x2F`):
```json
"{": "40,00,2f"
```

### Recommendation:
Someone with a physical Swiss-German keyboard should generate a fresh `ch.json` using a HID keycode scanner. The quickest approach is:
1. Use [USB HID Keyboard Scan Codes](https://gist.github.com/MightyPork/6da26e382a7ad91b5496ee55fdc73db2) as reference
2. Walk through every key + modifier combo (normal, Shift, AltGr, Shift+AltGr)
3. Submit a corrected PR

I have the tooling to help validate a corrected file if someone can provide the raw keycode mappings.

**Comment by bad-antics** [comment]
This is a known issue with the Swiss-German (ch.json) HID language file. The root causes are:

### 1. Numpad vs Regular Key Priority
The language file maps digits to **numpad keycodes** instead of the number row. NumLock must be ON for numbers to work, which is not guaranteed.

**Fix** — replace numpad keycodes with number row codes:
| Char | Wrong (numpad) | Correct (number row) |
|------|---------------|----------------------|
| 1 | 00,00,59 | 00,00,1e |
| 2 | 00,00,5a | 00,00,1f |
| ... | ... | ... |
| 0 | 00,00,62 | 00,00,27 |

### 2. Backslash via AltGr
On CH layout, backslash is AltGr + < (key left of Y). The correct HID mapping:
```
"\\": "40,00,64"
```
(0x40 = Right Alt modifier, 0x64 = <> key)

### 3. Curly brace {
CH layout: AltGr + u-umlaut (keycode 0x2F):
```
"{": "40,00,2f"
```

### Recommendation:
The cleanest fix is regenerating ch.json using a HID keycode scanner. Walk through every key with normal, Shift, AltGr, and Shift+AltGr modifiers. Reference: https://gist.github.com/MightyPork/6da26e382a7ad91b5496ee55fdc73db2

I can help validate a corrected file if someone provides raw keycode mappings from a physical CH keyboard.
